### ISSUE TITLE
[Bug]: Unreproducible builds with `experimental.strictExecutionOrder`

### ISSUE THREAD

**Issue by thesmartwon** [rolldown/rolldown#2655]
### Reproduction link or steps

https://github.com/thesmartwon/rolltest/tree/renamed-exports

```sh
npm i
for i in $(seq 1 20); do npm run build; done
```

### What is expected?

The same vendor chunk between builds.

### What is actually happening?

Multiple different vendor chunks:
```sh
> ls dist
entry.css  entry.js  vendor-ott1P2uq.js  vendor-vG_PKRmH.js
```

```diff
diff dist/vendor-ott1P2uq.js dist/vendor-vG_PKRmH.js
4830c4830
< export { __esm, createBroadcastChannelSynchronizer, createMergeableStore, init_index_client, init_with_schemas$1 as init_with_schemas, init_with_schemas as init_with_schemas$1, mount };
\ No newline at end of file
---
> export { __esm, createBroadcastChannelSynchronizer, createMergeableStore, init_index_client, init_with_schemas, init_with_schemas$1, mount };
\ No newline at end of file
```

### System Info

```Shell
Rolldown 0.14.0
```

### Any additional comments?

Feels like a bug with `experimental.strictExecutionOrder`.

**Comment by hyf0** [comment]
I check `0.14.0-snapshot-12d7e71-20241201004055` version and it seems fine. Could you check again with newest rolldown version?

**Comment by thesmartwon** [comment]
I should not have committed https://github.com/thesmartwon/rolltest/commit/f4ea286d7c0bdb66dbf13b8663beb68004eadac6. I've reverted it and that snapshot version still creates unreprodicible builds:
```sh
~/src/rolltest(renamed-exports)$ ls dist
entry.css  entry.js  vendor-BrdO3rTv.js  vendor-CQOkENnk.js
```

**Comment by IWANABETHATGUY** [comment]
https://github.com/rolldown/rolldown/pull/3021/files#diff-3feddf24775cd72bfb153e4d4c76bcdf1eee88583ced7febfdbcd44075a45d2a
, not sure if they are same issue.

**Comment by IWANABETHATGUY** [comment]
https://github.com/rolldown/rolldown/pull/3021/commits/a296d74b57e1805b610c0170da0ed0d4dc090833, this patch seems worked for me, I will try to fix it in separate pr.

**Comment by hyf0** [comment]
@thesmartwon Could you try again with latest rolldown?

**Comment by thesmartwon** [comment]
Fixed in 0.15.0-snapshot-86db816-20241210003559. Thanks!
